### ISSUE TITLE
Rules: defaults and categories

### ISSUE THREAD

**Issue by slacktracer** [eslint/eslint#3387]
Is https://github.com/eslint/eslint/blob/master/conf/eslint.json supposed to be the default for all rules? 

If so, is it in any specific order? Would it benefit from being ordered by category as established on the rules page? (I'm doing this for myself and could share)

To show what I mean:

```javascript
{
    "ecmaFeatures": {},
    "rules": {

        "comma-dangle": [2, "never"],
        "no-cond-assign": 2,
        "no-console": 2,
        "no-constant-condition": 2,
        "no-control-regex": 2,
        "no-debugger": 2,
        "no-dupe-args": 2,
        "no-dupe-keys": 2,
        "no-duplicate-case": 2,
        "no-empty-character-class": 2,
        "no-empty": 2,
        "no-ex-assign": 2,
        "no-extra-boolean-cast": 2,
        "no-extra-parens": 0,
        "no-extra-semi": 2,
        "no-func-assign": 2,
        "no-inner-declarations": [2, "functions"],
        "no-invalid-regexp": 2,
        "no-irregular-whitespace": 2,
        "no-negated-in-lhs": 2,
        "no-obj-calls": 2,
        "no-regex-spaces": 2,
        "no-sparse-arrays": 2,
        "no-unreachable": 2,
        "use-isnan": 2,
        "valid-jsdoc": 0,
        "valid-typeof": 2,
        "no-unexpected-multiline": 0,

        "accessor-pairs": 0,
        "block-scoped-var": 0,
        "complexity": [0, 11],
        "consistent-return": 0,
        "curly": [0, "all"],
        "default-case": 0,
        "dot-location": 0,
        "dot-notation": [0, { "allowKeywords": true }],
        "eqeqeq": 0,
        "guard-for-in": 0,
        "no-alert": 0,
        "no-caller": 0,
        "no-div-regex": 0,
        "no-else-return": 0,
        "no-empty-label": 0,
        "no-eq-null": 0,
        "no-eval": 0,
        "no-extend-native": 0,
        "no-extra-bind": 0,
        "no-fallthrough": 2,
        "no-floating-decimal": 0,
        "no-implicit-coercion": 0,
        "no-implied-eval": 0,
        "no-invalid-this": 0,
        "no-iterator": 0,
        "no-labels": 0,
        "no-lone-blocks": 0,
        "no-loop-func": 0,
        "no-multi-spaces": 0,
        "no-multi-str": 0,
        "no-native-reassign": 0,
        "no-new-func": 0,
        "no-new-wrappers": 0,
        "no-new": 0,
        "no-octal-escape": 0,
        "no-octal": 2,
        "no-param-reassign": 0,
        "no-process-env": 0,
        "no-proto": 0,
        "no-redeclare": 2,
        "no-return-assign": 0,
        "no-script-url": 0,
        "no-self-compare": 0,
        "no-sequences": 0,
        "no-throw-literal": 0,
        "no-unused-expressions": 0,
        "no-useless-call": 0,
        "no-void": 0,
        "no-warning-comments": [0, { "terms": ["todo", "fixme", "xxx"], "location": "start" }],
        "no-with": 0,
        "radix": 0,
        "vars-on-top": 0,
        "wrap-iife": 0,
        "yoda": [0, "never"],

        "strict": 0,

        "init-declarations": 0,
        "no-catch-shadow": 0,
        "no-delete-var": 2,
        "no-label-var": 0,
        "no-shadow-restricted-names": 0,
        "no-shadow": 0,
        "no-undef-init": 0,
        "no-undef": 2,
        "no-undefined": 0,
        "no-unused-vars": [2, {"vars": "all", "args": "after-used"}],
        "no-use-before-define": 0,

        "callback-return": 0,
        "handle-callback-err": 0,
        "no-mixed-requires": [0, false],
        "no-new-require": 0,
        "no-path-concat": 0,
        "no-process-exit": 0,
        "no-restricted-modules": 0,
        "no-sync": 0,

        "array-bracket-spacing": [0, "never"],
        "brace-style": [0, "1tbs"],
        "camelcase": 0,
        "comma-spacing": 0,
        "comma-style": 0,
        "computed-property-spacing": [0, "never"],
        "consistent-this": [0, "that"],
        "eol-last": 0,
        "func-names": 0,
        "func-style": [0, "declaration"],
        "id-length": 0,
        "id-match": 0,
        "indent": 0,
        "key-spacing": [0, { "beforeColon": false, "afterColon": true }],
        "lines-around-comment": 0,
        "linebreak-style": [0, "unix"],
        "max-nested-callbacks": [0, 2],
        "new-cap": 0,
        "new-parens": 0,
        "newline-after-var": 0,
        "no-array-constructor": 0,
        "no-continue": 0,
        "no-inline-comments": 0,
        "no-lonely-if": 0,
        "no-mixed-spaces-and-tabs": [2, false],
        "no-multiple-empty-lines": [0, {"max": 2}],
        "no-nested-ternary": 0,
        "no-new-object": 0,
        "no-spaced-func": 0,
        "no-ternary": 0,
        "no-trailing-spaces": 0,
        "no-underscore-dangle": 0,
        "no-unneeded-ternary": 0,
        "object-curly-spacing": [0, "never"],
        "one-var": 0,
        "operator-assignment": [0, "always"],
        "operator-linebreak": 0,
        "padded-blocks": 0,
        "quote-props": 0,
        "quotes": [0, "double"],
        "semi-spacing": [0, {"before": false, "after": true}],
        "semi": 0,
        "sort-vars": 0,
        "space-after-keywords": [0, "always"],
        "space-before-blocks": [0, "always"],
        "space-before-function-paren": [0, "always"],
        "space-in-parens": [0, "never"],
        "space-infix-ops": 0,
        "space-return-throw-case": 0,
        "space-unary-ops": [0, { "words": true, "nonwords": false }],
        "spaced-comment": 0,
        "wrap-regex": 0,

        "arrow-parens": 0,
        "arrow-spacing": 0,
        "constructor-super": 0,
        "generator-star-spacing": 0,
        "no-class-assign": 0,
        "no-const-assign": 0,
        "no-this-before-super": 0,
        "no-var": 0,
        "object-shorthand": 0,
        "prefer-const": 0,
        "prefer-spread": 0,
        "prefer-reflect": 0,
        "require-yield": 0,


        "max-depth": [0, 4],
        "max-len": [0, 80, 4],
        "max-params": [0, 3],
        "max-statements": [0, 10],
        "no-bitwise": 0,
        "no-plusplus": 0,

        "prefer-template": 0

    }
}
```

Note that `prefer-template` is not on the rules page yet.

Thank you very much.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by nzakas** [comment]
Yes, these are the defaults. We have them ordered in alphabetical order so rules are easy to find (separated into two groups - those beginning with "no" and all the others). This file isn't intended to be a guide for end users, so reorganizing by category isn't preferable. We'd much rather point people to the rule documentation for that information.

**Comment by slacktracer** [comment]
Great!

Thank you, @nzakas!
